File: src/test/java/durr_vie/CharacterStrings/Anagrams/test_anagrams.py
from src.main.java.durr_vie.CharacterStrings.Anagrams.anagrams import anagrams
from typing import List
import pytest


test_data = [
    ("below the car is a rat drinking cider and bending its elbow while this thing is an arc that can act like a cat which cried during the night caused by pain in its bowel1", [
     ['bowel', 'below', 'elbow'], ['arc', 'car'], ['night', 'thing'], ['cried', 'cider'], ['act', 'cat']])
]


@ pytest.mark.parametrize("input,expected_result", test_data)
def test_anagrams(input: str, expected_result: List[str]):
    assert expected_result == anagrams(input)

File: src/test/java/durr_vie/CharacterStrings/Anagrams/test_anagrams.rb
require 'test/unit'
require 'rspec'
require 'rspec/autorun'
require './src/main/java/durr_vie/CharacterStrings/Anagrams/anagrams.rb'

=begin
RSpec.describe 'anagrams' do
  describe '#some_method_under_test' do
    subject { anagrams input }

    [
      ["below the car is a rat drinking cider and bending its elbow while this thing is an arc that can act like a cat which cried during the night caused by pain in its bowel", [
     ['bowel', 'below', 'elbow'], ['arc', 'car'], ['night', 'thing'], ['cried', 'cider'], ['act', 'cat']]]
    ].each do |input, expected_output|
      it "when the input is #{input}" do
        # let(:input_value) { input }
        actual_result = anagrams(input)
        expect(anagrams(input)).to eql expected_output
        expect(actual_result).to equal(expected_output)
        expect(actual_result).to be == expected_output
      end
    end
  end
end
=end
